api/base/filters.py

```diff
@@ -44,36 +78,19 @@ class ODMFilterMixin(object):
 
     # TODO Handle simple and complex non-standard fields
 
-    TRUTHY = set(['true', 'True', 1, '1'])
-    FALSY = set(['false', 'False', 0, '0'])
-    DEFAULT_OPERATOR = 'eq'
-
     # For the field_comparison_operators, instances can be a class or a tuple of classes
-    field_comparison_operators = [
-        {
-            'field_type': ser.CharField,
-            'comparison_operator': 'icontains'
-        },
-        {
-            'field_type': ser.ListField,
-            'comparison_operator': 'in'
-        }
-    ]
+    field_comparison_operators = {
+        ser.CharField: 'icontains',
+        ser.ListField: 'in',
+    }
 
     def get_comparison_operator(self, key):
-
-        for operator in self.field_comparison_operators:
-            if isinstance(self.serializer_class._declared_fields[key], operator['field_type']):
-                return operator['comparison_operator']
+        for field_type, comparison_operator in self.field_comparison_operators.items():
```

Comment:
Actually, no, these changes make no sense to me. So the field_type field from the previous implementation had the capability of being a tuple rather than a single item. I can certainly have different items for each of the fields, but I don't see what you're imagining by way of determining the instance of the declared field. 

So either you are being more clever than I, which is certainly possible, or you're not seeing the unimplemented stuff, because the simple case is all we've needed so far. I won't just be checking the last class that we are comparing to. If someone subclasses CharField, for example, and it doesn't require a different comparison operator, then I shouldn't have to add another field type for the subclass into the field comparison operators. 

So, given that, does your suggestion make sense and, if so, can you give me a hint about what the if statement should be replaced with?